docs/web3.eth.rst

```diff
@@ -1076,6 +1076,10 @@ The following methods are available on the ``web3.eth`` namespace.
     The ``transaction`` and ``block_identifier`` parameters are handled in the
     same manner as the :meth:`~web3.eth.Eth.send_transaction()` method.
 
+    The ``state_override`` parameter allows gas estimation for cases when there
```

Comment:
Just curious, where did this description come from? It seems really specific, which may indeed be useful for that specific case but It seems like it may be more broadly used and might be better to describe it as simple overriding parts of the state before estimating the gas. Thoughts?

Reply:
You're right, I pulled this from the example they provided in the [original issue in geth](https://github.com/ethereum/go-ethereum/issues/27800). Happy to change it so that it doesnt include the specific transaction call examples.

Would this be better? Feel free to call out your own idea or modify my words.
```
The ``state_override`` parameter is useful when there is a chain of transaction calls. It overrides state so that the gas estimate of a transaction is accurate in cases where prior calls produce side effects.
```

Reply:
That makes more sense to me in a broader scope 👍🏼. Ship it.